microhapulator/data/marker_details_template.html

```diff
@@ -0,0 +1,183 @@
+                </tr>
+                {% endfor %}
+            </table>
+            <br><br><br>
+            <table>
+                <h4>Haplotype Calling</h4>
+                <tr>
+                    <th>Sample</th>
+                    <th class="alnrt">Typed Reads</th>
+                    <th class="alnrt">Total Reads</th>
+                    <th class="alnrt">Typing Rate</th>
+
+                </tr>
+                {% for sample, sample_data in typing_rates.items()%}
+                <tr>
+                    <td>{{sample}}</td>
+                    <td class="alnrt">{{ "{:,.0f}".format(sample_data.loc[markername, 'TypedReads']) }}</td>
+                    <td class="alnrt">{{ "{:,.0f}".format(sample_data.loc[markername, 'TotalReads'])}}</td>
+                    <td class="alnrt">{{ "{:.2f}".format(sample_data.loc[markername, 'TypingRate'] * 100)}}%</td>
+                </tr>
+                {% endfor %}
+            </table>
+        </div>
+        {% endfor %}
+    </div>
+
+</body>
+<script>
+
+    var urlParams = new URLSearchParams(window.location.search);
+    initMarkerDetails(urlParams.get('marker'))
+
+    function initMarkerDetails(marker) {
+        if (marker != null) {
+            show_selected_details(marker)
+        } else {
+            first_marker = document.getElementsByClassName("marker_details")[0]
+            show_selected_details(first_marker.id)
+        }
+    }
+
+    function show_selected_details(markername) {
+        var marker_details = document.getElementsByClassName("marker_details");
+        var found_marker = false;
+        for (var i = 0; i < marker_details.length; i++) {
+            if (marker_details[i].id == markername) {
+                found_marker = true
+                colorOffsets(marker_details[i])
+                document.getElementById('drop_down').value = markername
+                marker_details[i].style.display = "block";
+            } else {
+                marker_details[i].style.display = "none";
+            }
+        }
+        if (found_marker == false) {
+            alert(markername + " not found");
+        }
+    }
+
+    function colorOffsets(marker) {
+        var sequence = marker.getElementsByClassName("sequence")[0].innerHTML
+        var offsets = marker.getElementsByClassName("offsets")[0].innerHTML.split(", ");
+        var colored_sequence = "";
+        for (var j = 0; j < sequence.length; j++) {
+            if (offsets.includes(j.toString())) {
+                colored_sequence += '<span style="color:red' + '">' + sequence[j] + '</span>';
+            } else {
+                colored_sequence += sequence[j];
+            }
+        }
+        marker.getElementsByClassName("sequence")[0].innerHTML = colored_sequence;
+    }
+
+    function search_marker() {
+        var input = document.getElementById('searchbar').value;
+        show_selected_details(input);
+        document.getElementById('drop_down').value = input;
+    }
+
+</script>
```

Comment:
I must say, this is very clean and concise JavaScript code for someone who has been programming JavaScript for less than a month!